src/utilities/resolveSchemaCoordinate.ts

```diff
@@ -0,0 +1,313 @@
+import { inspect } from '../jsutils/inspect.js';
+
+import type {
+  ArgumentCoordinateNode,
+  DirectiveArgumentCoordinateNode,
+  DirectiveCoordinateNode,
+  MemberCoordinateNode,
+  SchemaCoordinateNode,
+  TypeCoordinateNode,
+} from '../language/ast.js';
+import { Kind } from '../language/kinds.js';
+import { parseSchemaCoordinate } from '../language/parser.js';
+import type { Source } from '../language/source.js';
+
+import type {
+  GraphQLArgument,
+  GraphQLEnumValue,
+  GraphQLField,
+  GraphQLInputField,
+  GraphQLNamedType,
+} from '../type/definition.js';
+import {
+  isEnumType,
+  isInputObjectType,
+  isInterfaceType,
+  isObjectType,
+} from '../type/definition.js';
+import type { GraphQLDirective } from '../type/directives.js';
+import type { GraphQLSchema } from '../type/schema.js';
+
+/**
+ * A resolved schema element may be one of the following kinds:
+ */
+export interface ResolvedNamedType {
+  readonly kind: 'NamedType';
+  readonly type: GraphQLNamedType;
+}
+
+export interface ResolvedField {
+  readonly kind: 'Field';
+  readonly type: GraphQLNamedType;
```

Comment:
Is there a reason we're not being more specific? We do have assertions for this, I think?

```suggestion
  readonly type: GraphQLObjectType | GraphQLInterfaceType;
```

(May need `<unknown, unknown>` added.)

Reply:
I suppose this makes sense, `GraphQLUnionType` ∈ `GraphQLNamedType`, but `GraphQLUnionType` is not a valid type here.